### DISCUSSION TITLE
Extremely slow connection establish

### DISCUSSION THREAD

**Discussion by danielmarschall** [rustdesk/rustdesk#7422] [Q&A] [answered]
Hello,

I have a strange problem which I try to solve for many days, but didn't find any solution. I hope you can help me.

I have a custom server which is running the latest version of github.com/rustdesk/rustdesk-server

I have several Windows computers which have the latest nightly RustDesk version installed (I tried it with the stable 2.3.1-1 too).
In the network settings, the ID and Relay servers as well as the server public key is entered.

When I connect to 123 456 789 , then it takes 15-25 seconds until the login screen comes. But after that, everything is fast and working.

When I connect to 123 456 789 @ (servername here) .com , then the connection is established immediately.

This is weird because the client has "(servername here).com" already entered as ID and Relay server, so I would assume the information "@(servername here).com" should be applied implicitly when I just enter "123 456 789".  I do not understand why there is a difference in "123 456 789" and "123 456 789 @ (servername here).com"

I have this performance problem on a Windows 10 and a Windows 11 computer as well as a macintosh computer.
I don't have this problem on a Windows Server 2022 computer. There, "123 456 789" and "123 456 789 @ (servername here).com" are both fast. All these computers are in the same network, so an Internet problem is unlikely.

What can I do?

Thank you for your help!

**Comment by xlionjuan** [comment]
By default, RustDesk Client and RustDesk Server will try to let both establish direct connection, so it won't need relay server, it can reduce the load of the server, user can also benefits from lower latency.

And this test need times.
if you wanna bypass this, you can Checked the devices option that will force to connect to relay every new connection

Or on 1.2.4
If you're establishing first connection, add `/r` suffix at the end of the ID, for example, `12345678/r`

Or you wanna always using relay for all devices, go check RustDesk server documentation, there is a option to let every devices using relay for always.

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/discussions/5002#discussioncomment-6571860
